docs/DeveloperGuide.md

```diff
@@ -380,6 +380,59 @@ It is designed to preserve the Command Design Pattern. Through the implementatio
+
+#### How the feature is implemented
+The sort feature uses the `sort` command, prefix `by/` before the specified `SORT_KEY` and prefix `in/` before the specified `SORT_ORDER`.
+
+Given below is an example usage scenario of how the sort mechanism behaves at each step to sort the task:
+
+Step 1. User inputs `sort by/deadline in/asc` to sort the task list by the `Deadline` in the ascending order.
+    
+Step 2. Upon receiving the user's input, `LogicManager` calls `HarmoniaParser#parseCommand()` to parse the user input.
+
+Step 3. The first word of the user input is `sort`, which matches the command for `SortCommand`. This initialises `SortCommandParser`.
+
+Step 4. `SortCommandParser#parse()` is called. `SORT_KEY` with prefix `by/` and `SORT_ORDER` with prefix `in/` are extracted out as a `SortKey` and `SortOrder` objects.
+
+Step 5. A `SortCommand` is initialised using the `SortKey` and `SortOrder`. The `SortCommand` constructor creates a `Comparator` used for sorting the list. `SortCommand` is returned to `LogicManager` for execution.
+
+Step 6. After `SortCommand#execute()` is called, `model#updateSortedTaskList(Comparator)` is invoked to sort the task list using the created `Comparator`. The command result is returned and displayed to the user.
+
+#### Why it is implemented that way
+The underlying `UniqueTaskList` uses an `ObservableList` and therefore it allows the application listen to changes and render according specified requirements. One such abstraction used in the initial implementation is the `FilteredList`. With the `FilteredList` the application grants the ability to filter the list using a `Predicate`. Sorting was designed to be an extension of this concept using `SortedList` abstraction where the application can sort the list by a given `Comparator` and render its output instantaneously.
+
+#### Design considerations:
+**Aspect: How the `Comparator` is created:**
+* **Alternative 1 (current choice):** Use Factory design pattern.
+    * Pros:
+        * Easier to extend
+        * Abstracts out the complexity of creating a Comparator
+        * Cons:
+            * Makes the code verbose
+* **Alternative 2:** Create `Comparator` at the point of parsing.
+    * Pros:
+        * Less code needed to implement
+    * Cons:
+        * Leads to poor code quality `Comparator` creation becomes more complex
+
+**Aspect: How the sorted list is stored/rendered:**
+* **Alternative 1 (current choice):** Use `javafx.collections.transformation.SortedList`.
+    * Pros:
+        * Abstracts out the details of rendering
+        * Abstracts out the details behind sorting (i.e. only need to pass in the `Comparator`)
+        * Cons:
+            * Difficult to preserve the ordering if tasks are added or edited
+* **Alternative 2:** Sorts the tasks using a stream and repopulate the `TaskList`.
+    * Pros:
+        * Straightforward implementation
+    * Cons:
+        * Mutability may lead to errors
```

Comment:
Just a small suggestion, but maybe you can consider explaining the specific [problems](https://web.mit.edu/6.005/www/fa15/classes/09-immutability/#:~:text=Mutable%20objects%20reduce%20changeability,the%20code%20harder%20to%20change.) that comes with Mutability! i.e. [non-OOP](https://nus-cs2103-ay2122s2.github.io/website/se-book-adapted/chapters/oop.html#what)